packages/GA4Client/src/commerce-handler.js

```diff
@@ -136,10 +136,147 @@ CommerceHandler.prototype.logCommerceEvent = function (event) {
+    eventName,
+    eventAttributes
+) {
+    if (this.common.forwarderSettings.measurementId) {
+        eventAttributes.send_to = this.common.forwarderSettings.measurementId;
+    }
+
+    gtag('event', eventName, eventAttributes);
+};
+
+// Google previously had a CheckoutOption event, and now this has been split into 2 GA4 events - add_shipping_info and add_payment_info
+// Since MP still uses CheckoutOption, we must map this to the 2 events using custom flags.  To prevent any data loss from customers
+// migrating from UA to GA4, we will set a default `set_checkout_option` which matches Firebase's data model.
+CommerceHandler.prototype.logCheckoutOptionEvent = function (
+    event,
+    affiliation
+) {
+    try {
+        var customFlags = event.CustomFlags,
+            GA4CommerceEventType = customFlags[GA4_COMMERCE_EVENT_TYPE],
+            ga4CommerceEventParameters;
+
+        // if no custom flags exist or there is no GA4CommerceEventType, the user has not updated their code from using legacy GA which still supports checkout_option
+        if (!customFlags || !GA4CommerceEventType) {
+            console.error(
+                'Your checkout option event for the Google Analytics 4 integration is missing a custom flag for "GA4.CommerceEventType". The event was sent using the deprecated set_checkout_option event.  Review mParticle docs for GA4 for the custom flags to add.'
+            );
+        }
+
+        switch (GA4CommerceEventType) {
+            case ADD_SHIPPING_INFO:
+                ga4CommerceEventParameters = buildAddShippingInfo(
+                    event,
+                    affiliation
+                );
+                break;
+            case ADD_PAYMENT_INFO:
+                ga4CommerceEventParameters = buildAddPaymentInfo(
+                    event,
+                    affiliation
+                );
+                break;
+            default:
+                ga4CommerceEventParameters = buildCheckoutOptions(
+                    event,
+                    affiliation
+                );
+                break;
+        }
+    } catch (error) {
+        console.error(
+            'There is an issue with the custom flags in your CheckoutOption event. The event was not sent.  Plrease review the docs and fix.',
+            error
+        );
+        return false;
+    }
+
+    this.sendCommerceEventToGA4(
+        mapGA4EcommerceEventName(event),
+        ga4CommerceEventParameters
+    );
+
+    return true;
+};
+
+CommerceHandler.prototype.logPromotionEvent = function (event) {
+    var self = this;
+    try {
+        var ga4CommerceEventParameters;
+        event.PromotionAction.PromotionList.forEach(function (promotion) {
+            ga4CommerceEventParameters = buildPromotion(promotion);
+
+            self.sendCommerceEventToGA4(
+                mapGA4EcommerceEventName(event),
+                ga4CommerceEventParameters
+            );
+            return true;
```

Comment:
this `return true` is happening within the `forEach`, so I believe that it will return and stop looping through.

You should move this 1 line lower, so that it happens after the `forLoop` completes